lib/rspec/core/filter_manager.rb

```diff
@@ -99,8 +99,10 @@ def prune_conditionally_filtered_examples(examples)
       # defined in the same file as the location filters. Excluded specs in
       # other files should still be excluded.
       def file_scoped_include?(ex_metadata, ids, locations)
-        no_location_filters = locations[ex_metadata[:absolute_file_path]].empty?
         no_id_filters = ids[ex_metadata[:rerun_file_path]].empty?
+        no_location_filters = locations[
+          File.expand_path(ex_metadata[:rerun_file_path])
+        ].empty?
```

Comment:
@myronmarston Ah yes that's a good point, but I can't think of a practical use case that relies on that behaviour.

The closest I can imagine is running same inner example from a shared example group across multiple rerun files. You'd have to use a dummy locations i.e. -

`rspec shared.rb:5 includes_shared_1.rb:999 includes_shared_2.rb:999`

Of course, if you want to do that you'd just use a tag on the example you want to test.